src/ripple/nodestore/DatabaseShard.h

```diff
@@ -45,12 +45,17 @@ class DatabaseShard : public Database
         @param config The configuration for the database
         @param journal Destination for logging output
     */
-    DatabaseShard(std::string const& name, Stoppable& parent,
-        Scheduler& scheduler, int readThreads,
-        Section const& config, beast::Journal journal)
-        : Database(name, parent, scheduler, readThreads, journal)
+    DatabaseShard(
+        std::string const& name,
+        Stoppable& parent,
+        Scheduler& scheduler,
+        int readThreads,
+        Section const& config,
+        beast::Journal journal)
+        : Database(name, parent, scheduler, readThreads, config, journal)
     {
         get_if_exists<std::uint32_t>(config, "ledgers_per_shard", lps_);
+        genesisShardIndex_ = seqToShardIndex(earliestSeq());
```

Comment:
Does this (and the line above) share @nbougalis's concern of initializing a static member each time this constructor is called? Even if its guaranteed to only be called once now, that seems a fragile/hard to enforce requirement.

Reply:
The issue is a side effect of making the seq configurable. It definitely introduced some ugliness, I'll fix it.